src/workflow/DirectoryManager.py

```diff
@@ -0,0 +1,76 @@
+import shutil
+from pathlib import Path
+import streamlit as st
+import zipfile
+from io import BytesIO
+
+
+class DirectoryManager:
+    # Methods related to directory management
```

Comment:
What is the intent of the class?
Does it create some GUI elements (e.g., a download button)? Then consider breaking the zipping and creating the download button into two methods. One related to backend fucntionality e.g. zipDirectory and one for frontend e.g. addDownloadButtonForZipFile

Reply:
It should just handle everything related to managing directories (creating, removing, making sure they exist, zipping).
Good point, the download button will fit much better in the StreamlitUI class.